employee-rostering-frontend/src/store/roster/operations.ts

```diff
@@ -168,4 +206,50 @@ export const getShiftRosterFor: ThunkCommandFactory<RosterSliceInfo & { spotList
       lastCalledShiftRosterArgs = params;
       dispatch(actions.setShiftRosterIsLoading(false));
     });
-  };
\ No newline at end of file
+  };
+
+export const getCurrentAvailabilityRoster: ThunkCommandFactory<PaginationData, SetAvailabilityRosterIsLoadingAction | SetAvailabilityRosterViewAction> = pagination =>
+  (dispatch, state, client) => {
+    const tenantId = state().tenantData.currentTenantId;
+    dispatch(actions.setAvailabilityRosterIsLoading(true));
+    return client.get<KindaAvailabilityRosterView>(`/tenant/${tenantId}/roster/availabilityRosterView/current?p=${pagination.pageNumber}&n=${pagination.itemsPerPage}`).then(newAvailabilityRosterView => {
+      const availabilityRosterView = convertKindaAvailabilityRosterViewToAvailabilityRosterView(newAvailabilityRosterView);
+      dispatch(actions.setAvailabilityRosterView(availabilityRosterView));
+      lastCalledAvailabilityRoster = getCurrentAvailabilityRoster;
+      lastCalledAvailabilityRosterArgs = pagination;
+      dispatch(actions.setAvailabilityRosterIsLoading(false));
+    });
+  };
+
+export const getAvailabilityRoster: ThunkCommandFactory<RosterSliceInfo & { pagination: PaginationData }, SetAvailabilityRosterIsLoadingAction | SetAvailabilityRosterViewAction> = params =>
+  (dispatch, state, client) => {
+    const tenantId = state().tenantData.currentTenantId;
+    const fromDateAsString = moment(params.fromDate).format("YYYY-MM-DD");
+    const toDateAsString = moment(params.toDate).add(1, "day").format("YYYY-MM-DD");
+    dispatch(actions.setAvailabilityRosterIsLoading(true));
+    return client.get<KindaAvailabilityRosterView>(`/tenant/${tenantId}/roster/availabilityRosterView?` +
+    `p=${params.pagination.pageNumber}&n=${params.pagination.itemsPerPage}` +
+    `&startDate=${fromDateAsString}&endDate=${toDateAsString}`).then(newAvailabilityRosterView => {
+      const availabilityRosterView = convertKindaAvailabilityRosterViewToAvailabilityRosterView(newAvailabilityRosterView);
+      dispatch(actions.setAvailabilityRosterView(availabilityRosterView));
+      lastCalledAvailabilityRoster = getAvailabilityRoster;
+      lastCalledAvailabilityRosterArgs = params;
+      dispatch(actions.setAvailabilityRosterIsLoading(false));
+    });
+  };
+
+export const getAvailabilityRosterFor: ThunkCommandFactory<RosterSliceInfo & { employeeList: Employee[] }, SetAvailabilityRosterIsLoadingAction | SetAvailabilityRosterViewAction> = params =>
+  (dispatch, state, client) => {
+    const tenantId = state().tenantData.currentTenantId;
+    const fromDateAsString = moment(params.fromDate).format("YYYY-MM-DD");
+    const toDateAsString = moment(params.toDate).add(1, "day").format("YYYY-MM-DD");
+    dispatch(actions.setAvailabilityRosterIsLoading(true));
+    return client.post<KindaAvailabilityRosterView>(`/tenant/${tenantId}/roster/availabilityRosterView/for?` +
```

Comment:
Mixing different levels of abstraction. HTTP level (method, URL, encoding parameters) and application logic (I can't easily understand what's actually posted here and how it's related to getting an employee's availability).

Could you create a wrapper around the rest client that would have methods in domain language? For example:
```ts
client.get<RosterState>(`/tenant/${tenantId}/roster/state`);
```
change to
```ts
client.getRosterState(tenantId);
```
-> less verbose, less distraction with lower level stuff.

Reply:
The REST client would get extremely large if I were to wrap every single REST method in it. As I see it, the "operations" class serves as the wrapper (with the exception it writes to AppState instead of returning a value).